### ISSUE TITLE
Only Display Inlined Editor When Text Is Selected

### ISSUE THREAD

**Issue by timayhelen** [elegantthemes/Divi-Beta#404]
It would be better if the tooltip only appears when text was selecting, and not when simply including into a paragraph. The tooltip doesn't serve a function when writing, only when highlighting text.

**Comment by timayhelen** [comment]
@fikrirasyid Once you mouse down in a module, it's impossible to ever activate hover buttons, even if you mouse down in a different module.

**Comment by timayhelen** [comment]
@fikrirasyid Hmm, maybe not actually. Now I can't re-create it. There is also the issue that, once you click into a module's text area, there is no way to access hover buttons until you click into a different text area. I think we need to make it so that, if you click anywhere outside of a tincymce editable area, it removes your cursor from the text and allows you to trigger hover buttons again.

**Comment by fikrirasyid** [comment]
@timayhelen I'll take a look at it. It is indeed quite tricky to access hover button once the cursor is focusing on tinyMCE area. A simple click / mouseLeave to outside tinyMCE area doesn't release the cursor from tinyMCE which causes the popover cannot be accessed.

**Comment by SlavaET** [comment]
@fikrirasyid make sure to pull the latest version of 1848-front-end-builder branch, it includes the fix for the focus out issue in Inline editor. That issue may create additional issues for you, so be sure you're on the latest codebase.

**Comment by fikrirasyid** [comment]
@timayhelen I've pushed a fix that fixes the issue you're describing above (with several differences in details tho). The commit fixes the issue by the adjusting the behaviour of inline tinyMCE editor + popover + inline editor popover as follow:

1. performing mouseDown in inline editor without selecting any text: simply hover out then hover enter will brings up the popover

![sit-1](https://cloud.githubusercontent.com/assets/916442/18378960/328cb6be-769a-11e6-982f-d5526cef13da.gif)

2. selecting text in inline editor: if user hasn't hover into the inline editor popover, simply hover out and enter like no. 1 will bring up the popover

![sit-2](https://cloud.githubusercontent.com/assets/916442/18378967/3fb74be2-769a-11e6-9463-4414a705ce69.gif)

3. selecting text in inline editor and user has hovering on the inline editor: click anywhere outside the module, then hover back to the module will bring up the popover

![sit-3](https://cloud.githubusercontent.com/assets/916442/18378975/47470eba-769a-11e6-802a-1e2816dc92a0.gif)

**Comment by timayhelen** [comment]
@fikrirasyid I guess I don't really like the idea of removing the Inline Editor popover when your mouse leaves the module. For small modules, it's quite easy to do by accident especially considering the popover will be outside the confines of the module.

Do you think there is a different approach we can take?

**Comment by fikrirasyid** [comment]
@timayhelen can you take a look at https://github.com/elegantthemes/submodule-builder/pull/1217?  I tried various approach and the mousetrap based approach to capture "click outside tinyMCE" appears to be best approach. Here's how it looks like:

![make-popover-available-on-inline-editor-leave](https://cloud.githubusercontent.com/assets/916442/18752941/22facb22-810e-11e6-9487-68895a2ebbd3.gif)

basically, if inline editor appears, single click outside tinymce will hide it and display module popovers.

***

I checked quite lots of alternative to this, and this appears to be the most stable alternative to hover-based check. Several other method that I've checked but appears to be unsustainable:

- Detect duration of mouseMove inside tinyMCE. This makes way too much assumption about what user intend to do. So no.
- Using inverse selector (select all except tinyMCE selector). The selector is really prone to error. There are way too much tinyMCE clicks that incorrectly detected
- Using tinyMCE's default focusOut and blur event. It has this uncomfortable method to determine blur. Basically: single click outside tinyMCE doesn't triggers blur. What triggers the blur: right click outside tinyMCE OR selecting text inside tinyMCE then single click outside tinyMCE. This can be easily perceived as error.

**Comment by fikrirasyid** [comment]
@timayhelen  I pushed an update to https://github.com/elegantthemes/submodule-builder/pull/1217 because I noticed this popover can appear on multiple modules if you do the following:

1. User clicks on different tinyMCE area
2. User clicks outside its module, then hovering to different module

![multiple module setting popovers](https://cloud.githubusercontent.com/assets/916442/18770415/c00fa828-815f-11e6-802d-88eb4e15f486.gif)

Forcing this behaviour doesn't work (automatically open popover when cursor is focused out tinyMCE area). What makes more sense is to avoid automatically opening popover when user clicks outside tinyMCE. So the expected gesture is:

1. User types with no popover
2. User clicks outside tinymce to focus out
3. User hovers back to display popover

![expected gesture](https://cloud.githubusercontent.com/assets/916442/18772981/27706556-8175-11e6-847f-05009c98597a.gif)
